internal/usagestats/event_handlers.go

```diff
@@ -221,3 +229,38 @@ func serializeLocalEvents(events []Event) ([]*database.Event, error) {
 
 	return databaseEvents, nil
 }
+
+// redactSensitiveInfoFromCloudURL redacts portions of URLs that
+// may contain sensitive info on Sourcegraph Cloud. We replace all paths,
+// and only maintain query parameters in a specified allowlist,
+// which are known to be essential for marketing analytics on Sourcegraph Cloud.
+func redactSensitiveInfoFromCloudURL(rawURL string) (string, error) {
+	parsedURL, err := url.Parse(rawURL)
+	if err != nil {
+		return "", err
+	}
+
+	parsedURL.RawPath = "/redacted"
+	parsedURL.Path = "/redacted"
+
+	marketingQueryParameters := map[string]string{
+		"utm_source":   "",
+		"utm_campaign": "",
+		"utm_medium":   "",
+		"utm_term":     "",
+		"utm_content":  "",
+	}
```

Comment:
This is more idiomatic Go for the use case here:

```suggestion
	marketingQueryParameters := map[string]struct{}{
		"utm_source":   {},
		"utm_campaign": {},
		"utm_medium":   {},
		"utm_term":     {},
		"utm_content":  {},
	}
```